simulator/examples/clock.rs

```diff
@@ -0,0 +1,196 @@
+fn draw_hour_hand(hour: u32) -> Line<BinaryColor> {
+    // Convert hour into a position around the circle in radians
+    let hour_radians = ((hour as f32 / 12.0) * 2.0 * PI) + START;
+
+    let hand_len = SIZE as f32 - 60.0;
+
+    let end = CENTER
+        + Point::new(
+            (hour_radians.cos() * hand_len) as i32,
+            (hour_radians.sin() * hand_len) as i32,
+        );
+
+    // Basic line hand
+    Line::new(CENTER, end).stroke(Some(BinaryColor::On))
+}
+
+/// Draw the minute hand (0-59)
+fn draw_minute_hand(minute: u32) -> Line<BinaryColor> {
+    // Convert minute into a position around the circle in radians
+    let minute_radians = ((minute as f32 / 60.0) * 2.0 * PI) + START;
+
+    let hand_len = SIZE as f32 - 30.0;
+
+    let end = CENTER
+        + Point::new(
+            (minute_radians.cos() * hand_len) as i32,
+            (minute_radians.sin() * hand_len) as i32,
+        );
+
+    // Basic line hand
+    Line::new(CENTER, end).stroke(Some(BinaryColor::On))
+}
+
+/// Draw digital clock just above center with black text on a white background
+///
+/// NOTE: The formatted time str must be passed in as references to temporary values in a
+/// function can't be returned.
+fn draw_digital_clock<'a>(time_str: &'a str) -> impl Iterator<Item = Pixel<BinaryColor>> + 'a {
+    let text = Font12x16::<BinaryColor>::render_str(&time_str)
+        .stroke(Some(BinaryColor::Off))
+        .translate(CENTER - Size::new(48, 48));
+
+    // Add some padding on the background around the time digits, here it's 2px in each direction
+    let offset = Size::new(2, 2);
+
+    let background = Rectangle::new(text.top_left() - offset, text.bottom_right() + offset)
+        .fill(Some(BinaryColor::On));
+
+    // Draw the white background first, then the black text. Order matters here
+    background.into_iter().chain(text)
+}
+
+fn main() {
+    let mut display = DisplayBuilder::new()
+        .title("Clock")
+        .size(DISP_SIZE as usize, DISP_SIZE as usize)
+        .scale(2)
+        .build_binary();
+
+    loop {
+        display.clear();
+
+        display.draw(draw_face());
+
+        let time = Local::now();
+
+        display.draw(draw_hour_hand(time.hour()));
+        display.draw(draw_minute_hand(time.minute()));
+        display.draw(draw_seconds_hand(time.second()));
+
+        // NOTE: In no-std environments, consider using
+        // [arrayvec](https://stackoverflow.com/a/39491059/383609) and a fixed size buffer
+        let clock_text = format!(
+            "{:02}:{:02}:{:02}",
+            time.hour(),
+            time.minute(),
+            time.second()
+        );
+
+        // Draw digital clock just above center
+        display.draw(draw_digital_clock(&clock_text));
+
+        // Draw a small circle over the hands in the center of the clock face. This has to happen
+        // after the hands are drawn so they're covered up
+        display.draw(
+            Circle::new(CENTER, 6)
```

Comment:
IMO a smaller circle looks better, maybe `r = 3`

Reply:
`r = 4` and a fill looks good too 🙂. I'm gonna go with that and call this bikeshed resolved haha

![2019-09-08](https://user-images.githubusercontent.com/800791/64494623-3dc13180-d287-11e9-87a6-dcaade9cd4e1.png)